src/os/bluestore/BlueStore.cc

```diff
@@ -802,6 +801,58 @@ void BlueStore::handle_conf_change(const struct md_config_t *conf,
       changed.count("bluestore_csum")) {
     _set_csum();
   }
+  if (changed.count("bluestore_compression") ||
+      changed.count("bluestore_compression_algorithm") ||
+      changed.count("bluestore_compression_min_blob_size") ||
+      changed.count("bluestore_compression_max_blob_size")) {
+    _set_compression();
+  }
+}
+
+void BlueStore::_set_compression()
+{
+  comp_min_blob_size = g_conf->bluestore_compression_min_blob_size;
+  comp_max_blob_size = g_conf->bluestore_compression_max_blob_size;
+
+  const char *alg = 0;
+  if (g_conf->bluestore_compression_algorithm == "snappy") {
+    alg = "snappy";
+  } else if (g_conf->bluestore_compression_algorithm == "zlib") {
+    alg = "zlib";
+  } else if (g_conf->bluestore_compression_algorithm.length()) {
```

Comment:
We can burden Compressor with the check for known compression algorithm instead of doing that here. This way adding new algorithm becomes transparent for the BlueStore